File: src/main/java/com/tram/springbootangularboard/domain/Account.java
package com.tram.springbootangularboard.domain;

import com.tram.springbootangularboard.dto.AccountDto;
import lombok.*;
import org.springframework.security.crypto.password.PasswordEncoder;

import javax.persistence.*;
import java.util.Arrays;
import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Builder
public class Account extends AuditorEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "account_email", unique = true, nullable = false)
    private String email;

    @Column(name = "account_password", nullable = false)
    private String password;

    @Column(name = "account_username", nullable = false)
    private String username;

    @Column(name = "account_role")
    @ElementCollection
    @Enumerated(value = EnumType.STRING)
    private List<UserRole> userRole = Arrays.asList(UserRole.ROLE_USER);

    @Column(name = "account_social_id")
    private String socialId;

    @Column(name = "account_social_profile_link")
    private String socialProfileLink;

    public Account(String email, String password, String username) {
        this.email = email;
        this.password = password;
        this.username = username;
    }

    public void encodePassword(PasswordEncoder passwordEncoder) {
        this.password = passwordEncoder.encode(this.password);
    }

    public static Account fromDto(AccountDto accountDto, PasswordEncoder passwordEncoder) {
        Account account = accountDto.toEntity();
        account.encodePassword(passwordEncoder);
        return account;
    }
}

File: src/main/java/com/tram/springbootangularboard/domain/Account.kt
package com.tram.springbootangularboard.domain

import com.tram.springbootangularboard.enum.UserRole
import javax.persistence.*

@Entity
@Table(name = "account")
data class Account(

        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        val id: Long? = null,

        @Column(name = "account_username")
        val username: String? = "김규남",

        @Column(name = "account_user_id")
        val userId: String? = "kyunam",

        @Column(name = "account_password")
        var password: String? = "123123",

        @Column(name = "account_role")
        @Enumerated(value = EnumType.STRING)
        val userRole: UserRole? = UserRole.USER,

        @Column(name = "account_social_id")
        val socialId: Long? = null,

        @Column(name = "account_profile_thumb_url")
        val profileThumbUrl: String? = null)